internal/cmd/util_unix.go

```diff
@@ -17,6 +21,54 @@ func fileInfoUID(info fs.FileInfo) int {
 	return int(info.Sys().(*syscall.Stat_t).Uid) //nolint:forcetypeassert
 }
 
+func darwinVersion() (map[string]any, error) {
+	if runtime.GOOS != "darwin" {
+		return nil, nil
+	}
+	darwinVersion := make(map[string]any)
+	for _, name := range []string{
+		"buildVersion",
+		"productName",
+		"productVersion",
+		"productVersionExtra",
+	} {
+		output, err := exec.Command("sw_vers", "--"+name).Output()
+		if err != nil {
+			return nil, err
+		}
+		darwinVersion[name] = strings.TrimSpace(string(output))
+	}
+
+	productVersion, ok := darwinVersion["productVersion"].(string)
```

Comment:
*Minor* suggestion here, but you can take it or leave it because I think that both approaches are readable. Instead of having a lookup on `productVersion` separate, add an `if name == "productVersion"` after line 39 where you do all of the conversion from line 44 to 68.

(If I were doing this, I would make the change and then see if it improves or worsens the readability; it could go either way by localizing the behaviour or by adding even more indenting. I think it's a coin flip as to which way it will go.)

Reply:
I prefer current version a bit more because this way two parts are clearly separated. In proposed version these parts are more entangled with each other and parsing logic which is unrelated for everything other than `productVersion` is one level deeper which worsens readbility a bit.